src/classes/AFFL_Affiliations_TDTM.cls

```diff
@@ -464,26 +477,45 @@ public class AFFL_Affiliations_TDTM extends TDTM_Runnable {
                     }
                 }
             }
+
+            //AFTER UNDELETE, update contact primary affiliation if blank, otherwise uncheck primary
+            if (triggerAction == TDTM_Runnable.Action.AfterUndelete) {
+                if (affl.npe5__Primary__c) {
+                    mapConIdAfflUndelete.put(affl.npe5__Contact__c, affl);
+                }
+            }
         }
 
-        // update contacts' primary affiliation field as appropriate
+        // update contacts' primary affiliation field
         set<id> setAllContactIds = new set<id>();
         setAllContactIds.addAll(mapConIdAccIdPrimary.keySet());
         setAllContactIds.addAll(mapConIdAccIdUnprimary.keySet());
+        setAllContactIds.addAll(mapConIdAfflUndelete.keySet());
 
         if (!setAllContactIds.isEmpty()) {
             list<Contact> conForUpdate = new list<Contact>();
             list<Contact> queryCon = [SELECT Id, Primary_Affiliation__c FROM Contact WHERE Id IN :setAllContactIds];
             
             for (Contact con : queryCon) {
-                if (mapConIdAccIdPrimary.containsKey(con.id)) {
+                if (mapConIdAccIdPrimary.containsKey(con.id) && con.Primary_Affiliation__c != mapConIdAccIdPrimary.get(con.id)) {
```

Comment:
Yep, we were updating it regardless of whether it was needed it so I added the check to avoid changing last modified date for nothing.